packages/lesswrong/server/languageModels/autoTagCallbacks.ts

```diff
@@ -204,15 +204,25 @@ async function getTagBotAccount(context: ResolverContext): Promise<DbUser|null>
   return account;
 }
 
-let tagBotUserIdCache: Promise<{id: string|null}>|null = null;
+let tagBotUserIdPromise: Promise<void>|null = null;
+// If undefined, hasn't been fetched yet; if null, the account doesn't exist.
+let tagBotUserId: string|null|undefined = undefined;
+
 export async function getTagBotUserId(context: ResolverContext): Promise<string|null> {
-  if (!tagBotUserIdCache) {
-    tagBotUserIdCache = (async () => {
-      const tagBotAccount = await getTagBotAccount(context);
-      return {id: tagBotAccount?._id ?? null};
-    })();
+  if (tagBotUserId === undefined) {
+    if (!tagBotUserIdPromise) {
+      tagBotUserIdPromise = new Promise((resolve) => {
+        void (async () => {
+          const tagBotAccount = await getTagBotAccount(context);
+          tagBotUserId = tagBotAccount?._id ?? null;
+          tagBotUserIdPromise = null;
+          resolve();
+        })();
+      });
+    }
+    await tagBotUserIdPromise
   }
```

Comment:
Would be good to have this documented, I think.  Just to confirm I understand why it's doing what it's doing: to avoid retaining a resolver context captured by the promise (formerly) stored in `tagBotUserIdCache`, we now instead set `tagBotUserIdPromise` to null at the end of the execution of its own promise, to both preserve the "caching" behavior (any calls to `getTagBotUserId` while the promise is executing for the first time will skip L213-222 and just hit `await tagBotUserIdPromise`) and also avoid retaining anything captured by the promise for any future executions, since then we just return `tagBotUserId`?

Reply:
Yep. Adding a comment to that effect.